### PR TITLE
[repo] Add Linux ARM64 to actions CI

### PR DESCRIPTION
**Pull request by ysolomchenko**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet/issues/1859

## Changes

Add otel-linux-arm64 to actions CI

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: .github/workflows/Component.BuildTest.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ on:
         required: false
         type: string
       os-list:
-        default: '[ "windows-latest", "ubuntu-latest" ]'
+        default: '[ "windows-latest", "ubuntu-latest", "otel-linux-arm64" ]'
```
Comment (xiang17) [inline .github/workflows/Component.BuildTest.yml:23]: Where do you get this OS "otel-linux-arm64"? I didn't find it with a simple search.

Comment (Kielek) [reply .github/workflows/Component.BuildTest.yml:23]: @xiang17, these runners are used in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/. No problems so far.

**Comment by ysolomchenko** [comment]
@reyang, there is also the possibility of adding Windows ARM to our CI. Should we do it?

**Review by Kielek — APPROVED** [review]

**Comment by reyang** [comment]
> @reyang, there is also the possibility of adding Windows ARM to our CI. Should we do it?

Yeah, that would be nice! (we're seeing more users on mobile, ARM64 servers, etc.)

**Pull request merged**